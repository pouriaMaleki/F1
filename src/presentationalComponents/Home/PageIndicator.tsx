import React from "react";
import css from "./PageIndicator.module.sass";
import classNames from "classnames";

interface IProps {
  count: number;
  selected: number;
  onSelect: any;
}

const PageIndicator = (props: IProps) => {
  // initialize new array with undefined values
  // to use "map" easily to render circles
  let pages: Array<undefined> = new Array(props.count);
  pages.fill(undefined);

  return (
    <div className={css.root}>
      {pages.map((_, index) => (
        <div
          key={index}
          className={classNames({
            [css.circle]: true,
            [css.selected]: props.selected === index
          })}
          onClick={() => props.onSelect(index)}
        />
      ))}
    </div>
  );
};

export { PageIndicator };
