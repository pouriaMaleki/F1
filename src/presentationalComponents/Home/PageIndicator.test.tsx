import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { PageIndicator } from "./PageIndicator";

describe("Render PageIndicator", () => {
  // snapshot test
  it("render correctly PageIndicator component", () => {
    const PageIndicatorComponent = renderer
      .create(<PageIndicator count={2} selected={1} onSelect={() => {}} />)
      .toJSON();
    expect(PageIndicatorComponent).toMatchSnapshot();
  });

  // test props
  it("render count of circles same as count prop", () => {
    const props = {
        count: 3,
        selected: 0,
        onSelect: () => {}
      },
      PageIndicatorComponent = mount(<PageIndicator {...props} />);

    expect(PageIndicatorComponent.find(".circle").length).toEqual(props.count);
  });

  it("render correct selected circle", () => {
    const props = {
        count: 3,
        selected: 1,
        onSelect: () => {}
      },
      PageIndicatorComponent = mount(<PageIndicator {...props} />);

    expect(
      PageIndicatorComponent.find(".circle")
        .at(props.selected)
        .hasClass("selected")
    ).toEqual(true);
  });

  // mock function props - function call test
  it("click on circle should call prop onSelect and pass clicked circle index as param", () => {
    const index = 2,
      onSelect = jest.fn(),
      props = {
        count: 3,
        selected: 1,
        onSelect
      },
      PageIndicatorComponent = mount(<PageIndicator {...props} />);
    PageIndicatorComponent.find(".circle")
      .at(index)
      .simulate("click");

    expect(onSelect).toHaveBeenCalledTimes(1);
    expect(onSelect.mock.calls[0][0]).toBe(index);
  });
});
