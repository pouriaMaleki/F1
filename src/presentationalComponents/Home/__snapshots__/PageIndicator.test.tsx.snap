// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PageIndicator render correctly PageIndicator component 1`] = `
<div
  className="root"
>
  <div
    className="circle"
    onClick={[Function]}
  />
  <div
    className="circle selected"
    onClick={[Function]}
  />
</div>
`;
